src/Schema/Types/LaravelEnumType.php

```diff
@@ -151,4 +151,13 @@ public function serialize($value): string
 
         return $key;
     }
+
+    public function parseValue($value)
+    {
+        if ($value instanceof $this->enumClass) {
+            return $value;
+        }
+
+        return parent::parseValue($value);
+    }
```

Comment:
Hey @stayallive, I think this might fix the issue at least for enums from `bensampo/laravel-enum`. I don't know what we can do about the enums modified by `@enum` tnough.